tests/test_username_api.py

```diff
@@ -40,306 +41,62 @@ def get_expected_response(website, user, status):
-
-      user = generate_random_username(15)
-      expected = get_expected_response('pinterest', user, 404)
-      actual = get_response(self.app, 'pinterest', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['instagram']['available_usernames'])
-  def test_available_instagram_username(self, user):
-    expected = get_expected_response('instagram', user, 404)
-    actual = get_response(self.app, 'instagram', user)
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(15)
-      expected = get_expected_response('instagram', user, 404)
-      actual = get_response(self.app, 'instagram', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['twitter']['available_usernames'])
-  def test_available_twitter_username(self, user):
-    expected = get_expected_response('twitter', user, 404)
-    actual = get_response(self.app, 'twitter', user)
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(15)
-      expected = get_expected_response('twitter', user, 404)
-      actual = get_response(self.app, 'twitter', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['facebook']['available_usernames'])
-  def test_available_facebook_username(self, user):
-    expected = get_expected_response('facebook', user, 404)
-    actual = get_response(self.app, 'facebook', user)
-
-    # this is special to facebook checking,
-    # and 'profile' field is only useful for
-    # frontend purposes. Skipping here.
-    del actual['profile']
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(25)
-      expected = get_expected_response('facebook', user, 404)
-      actual = get_response(self.app, 'facebook', user)
+  @parameterized.expand(load_test_cases('available_username'), testcase_func_name=custom_name_func)
```

Comment:
split like this: https://github.com/manu-chroma/username-availability-checker/blob/master/tests/test_get_avatar.py#L41